integration_tests/suite/helpers/fixtures.py

```diff
@@ -1,29 +1,46 @@
-# Copyright 2017 The Wazo Authors  (see the AUTHORS file)
+# Copyright 2017-2019 The Wazo Authors  (see the AUTHORS file)
 # SPDX-License-Identifier: GPL-3.0-or-later
 
+import copy
+import time
+import requests
 from functools import wraps
 from wazo_webhookd_client.exceptions import WebhookdError
+from xivo_test_helpers import until
 
 from .base import VALID_TOKEN
 
 
-def subscription(subscription_args):
+def subscription(subscription_args, track_test_name=True):
     '''This decorator is only compatible with instance methods, not pure functions.'''
     def decorator(decorated):
         @wraps(decorated)
         def wrapper(self, *args, **kwargs):
             webhookd = self.make_webhookd(VALID_TOKEN)
-            new_subscription = webhookd.subscriptions.create(subscription_args)
-            args = list(args) + [new_subscription]
-            try:
-                result = decorated(self, *args, **kwargs)
-            finally:
+
+            sub = copy.deepcopy(subscription_args)
+            if track_test_name:
+                # Add test name to help debugging
+                sep = "&" if "?" in sub['config']['url'] else "?"
+                sub['config']['url'] += sep + "test_case=" + decorated.__name__
+
+            new_subscription = webhookd.subscriptions.create(sub)
+            self.ensure_webhookd_consume_uuid(new_subscription['uuid'])
+
+            def cleanup():
```

Comment:
We usually use
```python
try:
    result = decorated(self, *args, **kwargs)
finally:
    # any cleanup code here
return result
```
I don't have any strong opinion about any of those methods. I'm curious to known why you prefer the addCleanup way of doing it and adopt that style in other fixtures.

Reply:
I don't have a strong opinion on that. If an exception is raise in this code block it will be reported differently by nose/pytest/... (error in test vs error in teardown).

I have reverted back to try/finally to stay like other fixture.